src/Mono.Android/Android.Runtime/AndroidRuntime.cs

```diff
@@ -388,52 +388,92 @@ public override void RegisterNativeMembers (JniType nativeClass, Type type, stri
 					return;
 				}
 
-				string[] members = methods!.Split ('\n');
-				if (members.Length < 2) {
+				int newLineIndex = methods!.IndexOf ('\n');
+				if (newLineIndex == -1) {
 					if (jniAddNativeMethodRegistrationAttributePresent)
 						base.RegisterNativeMembers (nativeClass, type, methods);
 					return;
 				}
 
-				JniNativeMethodRegistration[] natives = new JniNativeMethodRegistration [members.Length-1];
-				for (int i = 0; i < members.Length; ++i) {
-					string method = members [i];
-					if (string.IsNullOrEmpty (method))
-						continue;
-					string[] toks = members [i].Split (new[]{':'}, 4);
-					Delegate callback;
-					if (toks [2] == "__export__") {
-						var mname = toks [0].Substring (2);
-						MethodInfo? minfo = null;
-						foreach (var mi in type.GetMethods (BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static))
-							if (mi.Name == mname && JavaNativeTypeManager.GetJniSignature (mi) == toks [1]) {
-								minfo = mi;
-								break;
+				List<JniNativeMethodRegistration> natives = new List<JniNativeMethodRegistration> (16); // an initial capacity since most classes have more than 4-8 methods
+				ReadOnlySpan<char> methodsSpan = methods;
```

Comment:
If we're going for `ReadOnlySpan<char>`, we could probably further improve things by:

 1. Update `JNIEnv.RegisterJniNatives()` to instead create the `ReadOnlySpan<char>`: https://github.com/xamarin/xamarin-android/blob/85bfb77dc4f405026ab8b1fd08f3cd30deac2249/src/Mono.Android/Android.Runtime/JNIEnv.cs#L139
    This would avoid a `string` creation.

    ```csharp
    ReadOnlySpan<char> methods = new ReadOnlySpan<char>((void*) methods_ptr, methods_len);
    androidRuntime!.TypeManager.RegisterNativeMembers (jniType, type, methods);
    ```

 2. Overload this `RegisterNativeMembers()` method to take `ReadOnlySpan<char>`, and have the override invoke the overload:

    ```csharp
    public override void RegisterNativeMembers (JniType nativeClass, Type type, string? methods) =>
        RegisterNativeMembers (nativeClass, type, new ReadOnlySpan<char>(methods));
    public void RegisterNativeMembers (JniType nativeClass, Type type, ReadOnlySpan<char> methods)
    {
        …
    }
    ```

This would introduce complications around `FastRegisterNativeMembers()`, but those can be handled.